disttask/loaddata/scheduler.go

```diff
@@ -113,40 +121,44 @@ func (s *ImportScheduler) OnSubtaskFinished(ctx context.Context, subtaskMetaByte
 	logutil.BgLogger().Info("OnSubtaskFinished", zap.Any("taskMeta", s.taskMeta))
 	var subtaskMeta SubtaskMeta
 	if err := json.Unmarshal(subtaskMetaBytes, &subtaskMeta); err != nil {
-		return subtaskMetaBytes, err
+		return nil, err
 	}
 
-	dataEngine, ok := s.dataEngines.Load(subtaskMeta.ID)
+	val, ok := s.sharedVars.Load(subtaskMeta.ID)
 	if !ok {
-		return subtaskMetaBytes, errors.Errorf("data engine %d not found", subtaskMeta.ID)
+		return nil, errors.Errorf("sharedVars %d not found", subtaskMeta.ID)
 	}
-	engine, ok := dataEngine.(*backend.OpenedEngine)
+	sharedVars, ok := val.(*SharedVars)
 	if !ok {
-		return subtaskMetaBytes, errors.Errorf("data engine %d not found", subtaskMeta.ID)
+		return nil, errors.Errorf("sharedVars %d not found", subtaskMeta.ID)
 	}
-	closedEngine, err := engine.Close(ctx)
-	if err != nil {
-		return subtaskMetaBytes, err
+
+	logutil.BgLogger().Info("import data engine", zap.Any("id", subtaskMeta.ID))
+	if closedEngine, err := sharedVars.DataEngine.Close(ctx); err != nil {
+		return nil, err
+	} else if err := s.tableImporter.ImportAndCleanup(ctx, closedEngine); err != nil {
+		return nil, err
+	}
+
+	logutil.BgLogger().Info("import index engine", zap.Any("id", subtaskMeta.ID))
+	if closedEngine, err := sharedVars.IndexEngine.Close(ctx); err != nil {
+		return nil, err
+	} else if err := s.tableImporter.ImportAndCleanup(ctx, closedEngine); err != nil {
+		return nil, err
 	}
-	return subtaskMetaBytes, s.tableImporter.ImportAndCleanup(ctx, closedEngine)
+
+	sharedVars.mu.Lock()
```

Comment:
this lock is really needed?

Reply:
Actually it's not required, this is just to increase robustness.